.github/workflows/leetcode_directory_writer.yml

```diff
@@ -0,0 +1,39 @@
+# The objective of this GitHub Action is to update the leetcode DIRECTORY.md file (if needed)
+# when doing a git push
+name: leetcode_directory_writer
+on:
+  push:
+    paths:
+      - 'leetcode/src/**.cpp'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+      - name: Add python dependencies
+        run: |
+          pip install requests
+      - name: Write LeetCode DIRECTORY.md
+        run: |
+          python3 scripts/leetcode_directory_md.py 2>&1 | tee leetcode/DIRECTORY.md
+          git pull || true
```

Comment:
AFAIK, this is not needed if we use `fetch-depth` in `actions/checkout`.
CC: @tjgurwara99.

Reply:
Yeah this is not needed.